src/ios/CDVCamera.m

```diff
@@ -183,18 +183,35 @@ - (void)takePicture:(CDVInvokedUrlCommand*)command
     }];
 }
 
+// The key window (Apple's terminology) is the one displayed, given that all windows are on the same level number
+// Which they are in our app's case.
+UIWindow* currentKeyWindow = nil;
+
+- (void)gainFocus {
+    currentKeyWindow = [UIApplication sharedApplication].keyWindow;
+    [self.webView.window makeKeyAndVisible];
+
+}
+
+- (void)loseFocus {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if(currentKeyWindow != nil){
+            [currentKeyWindow makeKeyAndVisible];
+            currentKeyWindow = nil;
+        }
+    });
+}
+
 - (void)showCameraPicker:(NSString*)callbackId withOptions:(CDVPictureOptions *) pictureOptions
 {
+    CDVCameraPicker* cameraPicker = [CDVCameraPicker createFromPictureOptions:pictureOptions];
+    self.pickerController = cameraPicker;
+    cameraPicker.callbackId = callbackId;
+    // we need to capture this state for memory warnings that dealloc this object
+    cameraPicker.webView = self.webView;
```

Comment:
Why is this no longer done asynchronously?

Reply:
I've moved it back for safety's sake. Really it was taking stuff off the UI thread to lower the amount of blocking. My Objective-C isn't really good enough to properly work out wether this will introduce concurrency problems so, feel less"brave/foolhardy" this morning, I moved it back.